index.js

```diff
@@ -92,7 +110,23 @@ export default class Purchases {
       @param {RestoreTransactionsListener} restoreTransactionsListener_ Restore transactions listener 
   */
   static addRestoreTransactionsListener(restoreTransactionsListener_) {
-    restoreTransactionsListener.push(restoreTransactionsListener_);
+    if (typeof restoreTransactionsListener_ !== "function") {
```

Comment:
this is a nitpick, but it struck me as odd the inconsistency about using a trailing dash `_` for the other function listener arguments and not in the remove ones